### ISSUE TITLE
angr.vault.VaultShelf re-load confusion

### ISSUE THREAD

**Issue by bannsec** [angr/angr#1426]
Sounds like the solution might be a bit complicated, but attempting to re-open an existing vault didn't work. Specifically:

```python
angr.vaults.VaultShelf('/tmp/tmpzlzjgh62.db')
```

Errored out. Removing the .db did cause it to load. Maybe add extra load logic to handle this type of a case? I'm probably not the only one who wasn't aware of this strange behavior.

**Comment by ltfish** [comment]
Copy-pasted from angr Slack to provide extra context.

```
owlz [12:02 AM]
are we supposed to be able to re-open spiller databases?
```angr.vaults.VaultShelf('/tmp/tmpzlzjgh62.db')```
when directly previously that file was created by VaultShelf itself
it is then not able to open it or identify the file type

fish [12:16 AM]
@zardus ^^^
seems like a bug report :slightly_smiling_face:

owlz [12:18 AM]
hah, kk, i'm on a roll :wink:

zardus [12:19 AM]
ugh, freaking shelf

owlz [12:19 AM]
need an issue ticket?

zardus [12:19 AM]
yes please :slightly_smiling_face:

owlz [12:19 AM]
kk

zardus [12:19 AM]
the problem is, depending on your version of python, shelf will use different storage backends by default (edited) 
it might work if you leave out the `.db`
cause one of the backends appends that automatically
basically, using shelf was a terrible mistake

owlz [12:20 AM]
fuck
weird
yeah, that worked

zardus [12:20 AM]
:thumbsup:
in general, the way that vault is written is dumb, but the way that pickle works means we're kinda stuck with that

owlz [12:21 AM]
if endswith('.db') strip?

zardus [12:21 AM]
depending on the backend, that'll cause issues :slightly_smiling_face:

owlz [12:21 AM]
heh
```

**Comment by rhelmot** [comment]
Yeah, the reason for this (iirc) is actually that some backends store their
stuff in multiple files and will append a different extension to each. We
can probably work around it if there’s an interface for passing a file
descriptor to shelf?
